### DISCUSSION TITLE
Interaction object from Command Context

### DISCUSSION THREAD

**Discussion by tescher** [Snazzah/slash-create#351] [Q&A] [answered]
Is there a way for me to get the interaction object from the command context? I see the ID and token in there, but I need to object those refer to.

**Comment by Snazzah** [comment, marked as answer]
`ctx.data` gives the raw JSON data of any interaction.

**Reply by tescher** [reply]
Thx, I'll try it.

**Reply by tescher** [reply]
Looks like I need the actual interaction: Interaction<CacheType> object. I need it to display a modal using the now-standard tools. Is that available?

**Reply by Snazzah** [reply]
[`CommandContext#sendModal`](https://slash-create.js.org/#/docs/main/latest/class/CommandContext?scrollTo=sendModal)

**Reply by tescher** [reply]
OK, the modal appears after invoking from the commandContext, but how to I catch the submit? I see your example invoking a callback, but how do I catch it as an interaction using the custom_id?

**Reply by tescher** [reply]
I created an event called modalInteraction and registered it with a callback, but the event never happens. Is that the correct name?
